pkg/provisioner/ironic/ironic_test.go

```diff
@@ -134,8 +134,11 @@ func TestProvisionerIsReady(t *testing.T) {
 func TestGetUpdateOptsForNodeWithRootHints(t *testing.T) {
 
 	eventPublisher := func(reason, message string) {}
+	auth := clients.AuthConfig{Type: clients.NoAuth}
 
-	prov, err := newProvisioner(makeHost(), bmc.Credentials{}, eventPublisher)
+	prov, err := newProvisionerWithSettings(makeHost(), bmc.Credentials{}, eventPublisher,
+		"https://localhost", auth, "http://localhost", auth,
```

Comment:
The idea here is that these should never be called, right? Shouldn't we avoid using something real like localhost and instead specify e.g. `ironic.test`. (See [RFC 2606](https://tools.ietf.org/html/rfc2606#page-2).)

Reply:
Done